cozy/handle_tools.py

```diff
@@ -11,9 +12,11 @@
 from cozy.syntax_tools import fresh_var, mk_lambda
 from cozy.typecheck import is_collection
 
+## TODO: Write better documentation.  Merges each element of b into a and returns a.
 @typechecked
 def _merge(a : {THandle:Exp}, b : {THandle:Exp}) -> {THandle:Exp}:
     """NOTE: assumes ownership of `a`"""
+    ## TODO: Why does the code define a new variable rather than operating on a directly?
     res = a
```

Comment:
I am giving `a` a more useful name, since it is being repurposed as a result.
 - the rest of the function mutates the result `res` instead of the input `a`
 - this function can be made pure in an instant by changing the line to `res = dict(a)`